homeassistant/components/homekit/type_covers.py

```diff
@@ -141,19 +168,24 @@ def move_cover(self, value):
             else:
                 service, position = (SERVICE_CLOSE_COVER, 0)
 
-        self.hass.services.call(DOMAIN, service,
-                                {ATTR_ENTITY_ID: self.entity_id})
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        await self.hass.services.async_call(DOMAIN, service, params)
 
         # Snap the current/target position to the expected final position.
-        self.char_current_position.set_value(position)
-        self.char_target_position.set_value(position)
-        self.char_position_state.set_value(2)
+        self.hass.async_add_job(self.char_current_position.set_value, position)
```

Comment:
This is very inefficient calling 3 times async add job. They are all queued up but will also be picked up by 3 threads at the same time.